packages/bundler-plugin-core/src/utils/writeBundleHelper.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  type Output,
+  type ProviderUtilInputs,
+  type UploadOverrides,
+} from "../types";
+import { getPreSignedURL } from "./getPreSignedURL";
+import { type NormalizedOptions } from "./normalizeOptions";
+import { detectProvider } from "./provider";
+import { uploadStats } from "./uploadStats";
+
+interface WriteBundleHelperArgs {
+  options: NormalizedOptions;
+  output: Output;
+}
+
+export const writeBundleHelper = async ({
```

Comment:
yeah after seeing this helper, I'm increasingly convinced that a `output` class is a good idea.

then the this method would be called like `await output.write(options)` and you would ensure that `JSON.stringify(output)` works properly by overriding `toString()` method.

Reply:
Okay, moved over to an `Output` class, before i dive too deep into it's tests, can you let me know if I'm on the right path to what you're expecting?